### PR TITLE
feat/chunking

### PR DESCRIPTION
**Pull request by Flo-Pc**
## Requirements

- [x] I have read the `CODE_OF_CONDUCT.md`.
- [x] I have updated `CHANGELOG.md` & `tools/install/deb/changelog` to reflect my changes.

## Motivation

vSMTP was not compatible with the chunking feature as described in RFC #3030

### Added
- Added chunking support, including BDAT command support

### CODE REVIEW

File: src/vsmtp/vsmtp-protocol/src/reader.rs [hunk 1]
```diff
@@ -79,7 +79,39 @@ where
                 self.n = self.buffer.len();
             }
             loop {
-                if let Some(pos) = find(&self.buffer[..self.n], b"\r\n") {
+                if self.buffer.starts_with(b"BDAT ") {
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-protocol/src/reader.rs:82]: Something feels wrong

File: src/vsmtp/vsmtp-test/src/tests/protocol/bdat.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see https://www.gnu.org/licenses/.
+ *
+ */
+use crate::run_pipelined_test;
+
+// see https://datatracker.ietf.org/doc/html/rfc3030
+
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-test/src/tests/protocol/bdat.rs:20]: Add more tests without `pipelining`, and add assert on the content of the message received

File: src/vsmtp/vsmtp-protocol/src/command.rs [hunk 3]
```diff
@@ -427,6 +427,42 @@ impl TryFrom<UnparsedArgs> for RcptToArgs {
...
+    type Error = ParseArgsError;
+
+    fn try_from(UnparsedArgs(value): UnparsedArgs) -> Result<Self, Self::Error> {
+        let args_part = value
+            .iter()
+            .take_while(|b| b.is_ascii_digit())
+            .map(|b| char::from(*b))
+            .collect::<String>();
+
+        let size = args_part.trim().parse::<usize>().unwrap_or(0);
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-protocol/src/command.rs:449]: Should produce `InvalidArgs` error

**Review by Mathieu-Lala — CHANGES_REQUESTED** [review]

**Comment by ltabis** [comment]
Closing this since we release vSMTP v3.

**Pull request closed without merging**